ledger/sandbox/src/main/scala/com/digitalasset/platform/sandbox/stores/ledger/sql/dao/PostgresLedgerDao.scala

```diff
@@ -318,28 +318,21 @@ private class PostgresLedgerDao(
   // We therefore don't need to update anything if there is already some data for the given (contract, party) tuple.
   private val SQL_BATCH_INSERT_DIVULGENCES =
     """insert into contract_divulgences(contract_id, party, ledger_offset, transaction_id)
-      |values({contract_id}, {party}, {ledger_offset}, {transaction_id})
-      |on conflict on constraint contract_divulgences_idx
-      |do nothing""".stripMargin
+      |values({contract_id}, {party}, {ledger_offset}, {transaction_id})""".stripMargin
```

Comment:
This is statement should only insert the given row if it not already exists (otherwise you should get a unique constraint violation).
In DB2, you could try using `MERGE INTO` or `INSERT INTO ... WHERE NOT EXISTS`.